crates/radix-engine-toolkit-uniffi/src/transaction/manifest.rs

```diff
@@ -792,6 +793,20 @@ impl ExecutionSummary {
                     DetailedManifestClass::from_native(item, network_id)
                 })
                 .collect(),
+            newly_created_non_fungibles: native
+                .newly_created_non_fungibles
+                .into_iter()
+                .map(|item| {
+                    NonFungibleGlobalId::from_parts(
+                        Arc::new(Address::unsafe_from_raw(
+                            item.resource_address().into_node_id(),
+                            network_id,
+                        )),
+                        NonFungibleLocalId::from(item.local_id().clone()),
+                    )
+                    .unwrap()
```

Comment:
Let's avoid this unwrap by doing the following:

```rust
newly_created_non_fungibles: native
  .newly_created_non_fungibles
  .into_iter()
  .map(|item| {
      NonFungibleGlobalId::from_parts(
          Arc::new(Address::unsafe_from_raw(
              item.resource_address().into_node_id(),
              network_id,
          )),
          NonFungibleLocalId::from(item.local_id().clone()),
      )
  })
  .collect::<Result<_, _>>()?,
```

Reply:
This unwrap should never panic because we are converting valid `NativeNonFungibleGlobalId`, in normal operation this always should succeed. In case it fails it means `CoreExecutionSummary` is malformed so I think we should change `ExecutionSummary::from_native()` return type to `Result` and return error in that case. What is your opinion?

Reply:
Agreed that this is a path that should realistically never panic. However, since this function already returns a `Result` then it makes sense to propagate all `Result`s.